src/vs/workbench/api/node/extHostWindow.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import Event, { Emitter } from 'vs/base/common/event';
+import { IThreadService } from 'vs/workbench/services/thread/common/threadService';
+import { ExtHostWindowShape, MainContext, MainThreadWindowShape } from './extHost.protocol';
+
+export class ExtHostWindow implements ExtHostWindowShape {
+
+	private _proxy: MainThreadWindowShape;
+
+	private _onDidChangeWindowFocus = new Emitter<boolean>();
+	get onDidChangeWindowFocus(): Event<boolean> { return this._onDidChangeWindowFocus.event; }
```

Comment:
fyi - I started to use this pattern: `readonly onDidFoo:Event<T> = this._onDidDoo.event` - readonly ftw, saving a getter on the prototype, saving a function call at runtime 🐎